### PR TITLE
Add Error event in Blazor ViewModel (#3279)

### LINKED ISSUE #3279: Add an OnError Event to the Blazor ViewModel
**Is your feature request related to a problem? Please describe.**
I'd like to be able to react when a save fails and perform an action in my code when this happens. For example if a concurrency violation occurs during the save, I would like to pop up a modal and give the user an option to override the changes in the system or cancel them. While I'm currently doing this now with the ViewModelErrorText updating on the page, I don't see a way to trigger the modal when the ViewModelErrorText changes.

**Describe the solution you'd like**
I would like to see an Error event get invoked when an exception occurs during the save process. It could also be invoked when there is an error in the refresh as well.

**Describe alternatives you've considered**
I tried to potentially override the SaveAsync method on the ViewModel, but found myself rewriting the whole thing because I'm not able to set the Exception property in my inherited class, since it has a private setter.

### PR DESCRIPTION
**Pull request by rrrooommmaaa**
Close #3279

### CODE REVIEW

File: Source/Csla.Blazor.Test/ViewModelSaveAsyncErrorTests.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        ManageObjectLifetime = true,
+        Model = person,
+      };
+      SetupErrorHandler(vm);
+
+      // Act
+      await vm.SaveAsync();
+
+      // Assert
+      Assert.AreEqual("Inserted", TestResults.GetResult("FakePerson"));
```
Comment (rrrooommmaaa) [inline Source/Csla.Blazor.Test/ViewModelSaveAsyncErrorTests.cs:60]: There is problem using `TestResults` for this test. It's probably due to multiple `ConfigureAwait(false)` that `AsyncLocal` is sometimes addressed from different execution contexts.
I'd wire a dedicated variable in the test and a handler to update variable.
Thoughts?

**Review by rockfordlhotka — CHANGES_REQUESTED** [review]

File: Source/Csla.Blazor/ViewModel.cs [hunk 2]
```diff
@@ -284,6 +288,8 @@ public async Task SaveAsync()
       {
         HookChangedEvents(Model);
         IsBusy = false;
+        if (Exception != null)
+          Error?.Invoke();
```
[the remark this answers is not in the document]
Comment (rrrooommmaaa) [reply Source/Csla.Blazor/ViewModel.cs]: Fixed

**Review by rockfordlhotka — APPROVED** [review]

**Comment by rockfordlhotka** [comment]
@rrrooommmaaa thank you for the updates to the PR. We do still need your contributor agreement.

https://github.com/MarimerLLC/csla/blob/main/CONTRIBUTING.md

**Comment by rrrooommmaaa** [comment]
> https://github.com/MarimerLLC/csla/blob/main/CONTRIBUTING.md

I have just sent it to you via e-mail.

**Pull request merged**